src/ApplicationManager.cpp

```diff
@@ -55,18 +50,40 @@ void ApplicationManager::start() {
 
 /// Run everything with regard to behaviour trees
 void ApplicationManager::runOneLoopCycle() {
-    if (weHaveRobots && io::io.hasReceivedGeom) {
-        ai::analysis::GameAnalyzer::getInstance().start();
-        decidePlay(world, io::io.getField());
-        updateTrees();
-        updateCoaches();
-        runKeeperTree();
-        Status status = runStrategyTree();
-        this->notifyTreeStatus(status);
+
+    if (io::io.hasReceivedGeom) {
+        auto fieldMessage = io::io.getGeometryData().field();
+        auto worldMessage = io::io.getWorldState();
```

Comment:
This means world messages are updated every tick and not whenever a new one is available, right?

Reply:
exactly! This approach reduces the risk of race conditions or other weird bugs due to data changes while doing calculations.